setup.py

```diff
@@ -22,7 +22,7 @@
     packages=find_packages(exclude=["tests"]),
     zip_safe=False,
     include_package_data=True,
-    python_requires=">=3.6",
+    python_requires=">=3.7",
```

Comment:
Should we go straight to 3.8 min? Either way, would you mind putting this in its own PR so it's explict?